compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -17762,15 +17762,29 @@ TR::Node *fmaxminSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
          fmin = fmax = secondChild->getFloat();
       else
          {
-         if (firstChild->getFloat() <= secondChild->getFloat())
+         float first = firstChild->getFloat();
+         float second = secondChild->getFloat();
+         int nZeroBits = 0x80000000;
+         if (first <= second)
             {
-            fmin = firstChild->getFloat();
-            fmax = secondChild->getFloat();
+            if (first == second && first == 0)
```

Comment:
Wondering,  if GCC or other compiler adheres to IEE-754, then this condition would be incorrect right ?

Reply:
ah yes in that case it would not work. Changed it to compare as ints instead and put as first if condition (no nesting)

Reply:
So comparison of +0/-0 should return `true` if compiler is following IEEE-754. I asked the question to confirm this.

Reply:
Have not worked out on edge cases, can following check work ? 
```
if (first > second || (firstBits == pZeroBits && secondBits == nZeroBits))
   fmax = first
   fmin = second
else // first < second, first == second , (first == nZeroBits && second dontCare)
   fmax = second
   fmin = first
```

Reply:
yup that'll work. Still need the additional checks for NaNs though, since both max and min need to evaluate to the same NaN

Reply:
I think the previous implementation before your fix did that.